### PR TITLE
(MODULES-11458) Revert deferred function

### PR DESCRIPTION
**Pull request by Ramesh7**
## Summary

Revert deferred function

## Additional Context
Add any additional context about the problem here. 
- [ ] Root cause and the steps to reproduce. (If applicable)
- [ ] Thought process behind the implementation.

## Related Issues (if any)
Mention any related issues or pull requests.

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified. (For example `puppet apply`)

### CODE REVIEW

**Review by bastelfreak — CHANGES_REQUESTED** [review]
Can you please explain why this gets reverted?

File: lib/puppet/functions/user_password.rb [hunk 1]
```diff
@@ -1,12 +0,0 @@
-# frozen_string_literal: true
-
-Puppet::Functions.create_function(:user_password) do
-  dispatch :user_password do
-    param 'Variant[Sensitive[String], String]', :password
-    return_type 'Variant[Sensitive[String], String]'
-  end
-
-  def user_password(password)
-    password
```
[the remark this answers is not in the document]
Comment (Ramesh7) [reply lib/puppet/functions/user_password.rb:10]: yeah, it returns as it is. 
We have implemented based on earlier understanding and handover but I feel probably we are missing something here. That is one reason I am reverting.

Also there is customer reported issue where `/etc/shadow` is getting populting with `"Deferred({'name' => 'user_password', 'argume...` content.

[the remark this answers is not in the document]
Comment (peelman) [reply lib/puppet/functions/user_password.rb:10]: I am having this exact issue currently after updating the accounts module.  Rolling back in our environment until this is fixed.

```
cat /etc/shadow
...
admin:Deferred({'name' => 'user_password', 'arguments' => ['$6$OLbl..snip..gPrlY.']}):19943:0:99999:7:::
...
```

**Review by bastelfreak — APPROVED** [review]
the function isn't properly implemented, it only returns the password as plaintext. so I think it makes sense to revert this.

**Pull request merged**